controllers/reconcilers/grafana/ingress_reconciler.go

```diff
@@ -64,13 +64,17 @@ func (r *IngressReconciler) reconcileIngress(ctx context.Context, cr *v1beta1.Gr
 
 	// try to assign the admin url
 	if cr.PreferIngress() {
-		if len(ingress.Status.LoadBalancer.Ingress) > 0 {
-			ingress := ingress.Status.LoadBalancer.Ingress[0]
-			if ingress.Hostname != "" {
-				status.AdminUrl = fmt.Sprintf("https://%v", ingress.Hostname)
-			}
-			status.AdminUrl = fmt.Sprintf("https://%v", ingress.IP)
+		adminURL := r.getIngressAdminURL(ingress)
+
+		if len(ingress.Status.LoadBalancer.Ingress) == 0 {
+			return v1beta1.OperatorStageResultFailed, fmt.Errorf("ingress is not ready yet")
```

Comment:
we can return `OperatorStageResultInProgress` here, it's not an error if the Ingress is not ready

Reply:
Definitely makes sense, fixed.